### ISSUE TITLE
Server side fusioninventory -> native inventory documentation missing

### ISSUE THREAD

**Issue by AldarisPale** [glpi-project/glpi#14735]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0.7

### Bug description

Version 10.x brought native inventory capability, which is being advertised as compatible with FusionInventory.

What seem to be missing, are the instructions of how to migrate server side data about assets (including when was the last contact with the computer, last inventory date etc) to GLPI native inventory capability from fusioninventory plugin.

Please note the client side instructions (meaning agent running on a computer), can be found, which is good.

### Steps To reproduce

1. Use Glpi 9.x with fusioninventory plugin, have computers reporting to GLPI. Notice that you can see under computer when was the last contact with the agent, when was inventory last reported. Also notice that computer information being filled with data being reported by fusioninventory agent.
2. Migrate GLPI to 10.x, don't install fusioninventory plugin.
3. Notice that GLPI is reporting that autoinventory is being used, but nowhere can be see when was the last contact with the agent.
4. Also notice that there is lack of documentation about how to migrate GLPI itself from fusioninventory plugin to native inventory capability without data loss.

**Comment by AldarisPale** [comment]
If I check from version 9.x with fusioninventory plugin install, I will see the following information under computer in GLPI:
There's a "FusionInventory" section under computer (right after computer name, comment, network etc section) listing:
Agent:
Status:
FusionInventory tag
Public contact address:
Useragent:
**Last contact:
Last inventory:
Last boot:**

After migrating to 10.x (before uninstalling fusioninventory plugin):
There's no "Fusioninventory" section anymore and cannot find information where it's visible what are:
**Last contact:
Last inventory:
Last boot:**
Notice that there is robot icon with hover text "Automatic Inventory" next to computer name. Icon is not clickable.

After fusionplugin cleanup, there's still no "Fusioninventory section".
After disabling debug mode, there's still no "Fusioninventory section".

After switching manually to super-admin profile, I now see that there is "Inventory information" section under Computer section, but it says "Agent information is not available", although information was available before upgrading to 10.x.

**Comment by cedric-anne** [comment]
https://github.com/glpi-project/glpi-inventory-plugin will migrate agent `Last contact` and computers `Last inventory`.

`Last boot` is not currently migrated, but it can probably be added easilly.

**Comment by cedric-anne** [comment]
> So it should not be applicable and the functionality I'm looking for - last inventory date, last boot, etc should be available natively as I understand.

There is no migration command inside GLPI itself. This data will however be available as long as related machines will be inventoried by GLPI native inventory.

**Comment by AldarisPale** [comment]
Thanks, @cedric-anne !
Looks like https://github.com/glpi-project/glpi-inventory-plugin is not in functional state, as after trying to install version 1.2.1 on top of GLPI 10.0.7, I don't see plugin appearing under plugins and the following is being reported in php-errors.log:

```
glpiphplog.WARNING: *** PHP User Warning (512): plugin_version_glpi-glpiinventory method must be defined! in /var/www/html/src/Plugin.php at line 1743.
Backtrace:
src/Plugin.php:1743 trigger_error()
src/Plugin.php:1781 Plugin::getInfo()
src/Plugin.php:746 Plugin->getInformationsFromDirectory()
src/Plugin.php:517 Plugin->getNewInfoAndDirBasedOnOldName()
src/Plugin.php:497 Plugin->checkPluginState()
src/Plugin.php:229 Plugin->checkStates()
inc/includes.php:86 Plugin->init()
ajax/notifications_ajax.php:36 include()
public/index.php:73 require()
```

**Comment by AldarisPale** [comment]
Please disregard error messages in https://github.com/glpi-project/glpi/issues/14735#issuecomment-1557300624 - plugin was installed tonwrong directory. Correct directory name is glpiinventory.

After installing the plugin, I can now see that "Inventory information" has been populated under comouter with the following fields:
* Agents
* Inventory tag
* Public contact address
* Agent status
* Useragent
* Last inventory
* Request inventory

What seems to be missing are:
* Last boot
* Last contact
It also still says "Inventory file missing".

**Comment by AldarisPale** [comment]
Thanks, @stonebuzz .
After migrating from 9.x with fusioninventory plugin to 10.0.7 with glpiinventory plugin 1.2.1 and checking the exact computer I can see that:
* Last boot date is empty
* There's last inventory (and not last contact as on your screenshot)
* Last contact can be seen from agent information page

What are the steps that need to be taken (this is why I reported this issue in the first place) to get to the state which is shown on your screenshots?

If fusioninventory data is not being migrated, this would mean information loss.

**Comment by stonebuzz** [comment]
```Last boot``` is not migrated because is handled / updated at each inventory.

```last_inventory_update``` can be used from ```Search``` but is not displayed in computer form for now (need to be fixed)

![image](https://github.com/glpi-project/glpi/assets/7335054/a6ba3695-0f38-4eae-ac97-dbba42298df4)

> What are the steps that need to be taken (this is why I reported this issue in the first place) to get to the state which is shown on your screenshots?

Just run an inventory from one computer.

**Comment by trasher** [comment]
Community support is on a best effort basis; no fix date will be provided. If you want more, consider using professional support.

**Comment by AldarisPale** [comment]
@trasher for background, the need to migrate to 10.x before the end of June comes from https://glpi-project.org/glpi-9-5-x-will-be-discontinued/
I suspect we are not the only ones who have hit the same challenge.

**Comment by trasher** [comment]
https://github.com/glpi-project/glpi-inventory-plugin/pull/364 should fix missing data during migration.

**Comment by trasher** [comment]
Issue should have been fixed

**Comment by AldarisPale** [comment]
Thanks everyone involved, @trasher and @stonebuzz especially.

But - documentation/guide still seems to be missing.
For example, let say I'm running GLPI 9.5.12, using fusioninventory agents and plugin.
Now I want to migrate to 10.0.7, using native inventory.

What's the process?

Is it:
1. Download 10.0.7
2. install glpiinventory plugin
3. Enable native inventory at front/inventory.php
4. Run migrations

Or is it:
1. Download 10.0.7
2. Install fusioninventory plugin
3. Enable native inventory at fron/inventory.php
4. Run migrations

Or is it:
1. Download 10.0.7
2. Delete existing fusioninventory plugin
3. Run clean up via GLPI
4. Install glpiinventory plugin
5. Enable native inventory at fron/inventory.php
6. Run migrations

Asking, because there this seems to be undocumented, or atleast I'm unable to find documentation what should the correct approach be. If such documentation exists, please point me to it.

**Comment by trasher** [comment]
1st solution.
FusionInventoruy plugin is not compatible with native inventory and doe snot propose any migration.
Last solution is not correct: if you remove a plugin, you'll remove all its data, no migration would be available then.

**Comment by AldarisPale** [comment]
Sorry for delay.
Tried out the combination of GLPI development branch and glpi-plugin-inventory development branch.
Looks alot better, but "Last inventory update" is still empty after migration from glpi 9.x with fusioninventory.

**Comment by AldarisPale** [comment]
@stonebuzz 

### before migration:

```
glpi> select computers_id, last_fusioninventory_update from glpi_plugin_fusioninventory_inventorycomputercomputers LIMIT 2;
+--------------+-----------------------------+
| computers_id | last_fusioninventory_update |
+--------------+-----------------------------+
| 1            | 2023-05-30 07:10:04         |
| 2            | 2020-04-30 06:10:48         |
+--------------+-----------------------------+

2 rows in set
Time: 0.004s
glpi>

glpi> select last_inventory_update from glpi_computers where last_inventory_update is NULL;
(1054, "Unknown column 'last_inventory_update' in 'field list'")
```

### After installing GLPI 10

```
glpi> select last_inventory_update from glpi_computers where last_inventory_update is NULL;
+-----------------------+
| last_inventory_update |
+-----------------------+
| <null>                |
1140 rows in set
Time: 0.040s
glpi>

glpi> select last_inventory_update from glpi_computers where last_inventory_update is not NULL;
+-----------------------+
| last_inventory_update |
+-----------------------+
+-----------------------+

0 rows in set
Time: 0.004s
glpi>
```

### After activating plugins and removing fusioninventory plugin
```

glpi> select computers_id, last_fusioninventory_update from glpi_plugin_fusioninventory_inventorycomputercomputers LIMIT 2;
(1146, "Table 'glpi.glpi_plugin_fusioninventory_inventorycomputercomputers' doesn't exist")
glpi>

glpi> select last_inventory_update from glpi_computers where last_inventory_update is NULL;
+-----------------------+
| last_inventory_update |
+-----------------------+
| <null>                |
1140 rows in set
Time: 0.040s
glpi>

glpi> select last_inventory_update from glpi_computers where last_inventory_update is not NULL;
+-----------------------+
| last_inventory_update |
+-----------------------+
+-----------------------+

0 rows in set
Time: 0.004s
glpi>
```

**Comment by stonebuzz** [comment]
I don't know if this will be enough, but could you test this patch?
(keep the following PR https://github.com/glpi-project/glpi-inventory-plugin/pull/364)


```patch
diff --git a/install/update.native.php b/install/update.native.php
index a0f23c277..23b896088 100644
--- a/install/update.native.php
+++ b/install/update.native.php
@@ -300,7 +300,7 @@ function pluginGlpiinventoryUpdateNative($current_version, $migrationname = 'Mig
               FROM `glpi_plugin_glpiinventory_inventorycomputercomputers`
           ) AS `plugin_computers` ON `plugin_computers`.`computers_id` = `computers`.`id`
           SET
-          `computers`.`last_inventory_update` = `plugin_computers`.`last_inventory_update`
+          `computers`.`last_inventory_update` = `plugin_computers`.`last_fusioninventory_update`
           WHERE `computers`.`last_inventory_update` IS NULL;"
         );
 

```

**Comment by AldarisPale** [comment]
> I don't know if this will be enough, but could you test this patch? (keep the following PR [glpi-project/glpi-inventory-plugin#364](https://github.com/glpi-project/glpi-inventory-plugin/pull/364))
> 
> ```diff
> diff --git a/install/update.native.php b/install/update.native.php
> index a0f23c277..23b896088 100644
> --- a/install/update.native.php
> +++ b/install/update.native.php
> @@ -300,7 +300,7 @@ function pluginGlpiinventoryUpdateNative($current_version, $migrationname = 'Mig
>                FROM `glpi_plugin_glpiinventory_inventorycomputercomputers`
>            ) AS `plugin_computers` ON `plugin_computers`.`computers_id` = `computers`.`id`
>            SET
> -          `computers`.`last_inventory_update` = `plugin_computers`.`last_inventory_update`
> +          `computers`.`last_inventory_update` = `plugin_computers`.`last_fusioninventory_update`
>            WHERE `computers`.`last_inventory_update` IS NULL;"
>          );
>  
> ```

Hi and thanks. Tried it out, result during activation of glpiinventory plugin:
`- Error during the database query: UPDATE `glpi_computers` AS `computers` INNER JOIN ( SELECT `computers_id`, `last_inventory_update` FROM `glpi_plugin_glpiinventory_inventorycomputercomputers` ) AS `plugin_computers` ON `plugin_computers`.`computers_id` = `computers`.`id` SET `computers`.`last_inventory_update` = `plugin_computers`.`last_fusioninventory_update` WHERE `computers`.`last_inventory_update` IS NULL; - Error is Unknown column 'plugin_computers.last_fusioninventory_update' in 'field list'`

**Comment by trasher** [comment]
Field `last_fusioninventory_update` should have been renamed to `last_inventory_update` (see in `install/update.php`); therefore last change proposal is not correct.
This plugin update code is a real nightmare; but last error seems to confirm it's correctly renamed.

Please make sure there is a line in "`inventorycomputercomputers`" table with a properly filled inventory date before update.

**Comment by AldarisPale** [comment]
@trasher was the last update - https://github.com/glpi-project/glpi/issues/14735#issuecomment-1598172010 - for me?
If so, what should I check and how? I don't understand what is meant by 

> Please make sure there is a line in "inventorycomputercomputers" table with a properly filled inventory date before update.

or was your update addressed to @stonebuzz ?

**Comment by AldarisPale** [comment]
On another note - end of June 2023 is coming sooner than desired and that is when 9.x support is being dropped completely https://glpi-project.org/glpi-9-5-x-will-be-discontinued/
It would be most excellent if this problem could be fixed and the fixes would be incorporated into new releases of GLPI and glpiinventory plugin in order to migrate to released versions (as opposed to dev branches) before the end of June 2023 arrives.

**Comment by AldarisPale** [comment]
> > 9.x support is being dropped completely https://glpi-project.org/glpi-9-5-x-will-be-discontinued/
> 
> By the way, community support of the 9.x branch has already been stopped a year ago (GLPI 9.5.8 was the latest release that contained bug fixes), and GLPI 9.5.9 -> 9.5.13 were only containing security fixes backports.
> 
> The announcement you refer to talks about official support, ie support provided through a paid subscription. If you have no paid subscription, it should not change anything for you.

For how long will be security-fixes back-ported to 9.x, @cedric-anne ?

**Comment by AldarisPale** [comment]
What can I check/do on my end in order to progress with this problem?

**Comment by cedric-anne** [comment]
> For how long will be security-fixes back-ported to 9.x, @cedric-anne ?

Indeed, there will be no new security fixes for GLPI 9.5.

**Comment by trasher** [comment]
> What can I check/do on my end in order to progress with this problem?

As already said:
«Please make sure there is a line in "inventorycomputercomputers" table with a properly filled inventory date before update.»
We need an easy way to reproduce, but we do not have any data from plugin to test on.

Also, please keep current issue focused, 9.5 is discontinued fro a while as Cedric explained few days ago already.

**Comment by AldarisPale** [comment]
@trasher not sure I 100% understand your request, but if I run
`select computers_id, last_fusioninventory_update from glpi_plugin_fusioninventory_inventorycomputercomputers;`
before the migration, all rows have `last_fusioninventory_update` populated.
Example result is https://github.com/glpi-project/glpi/issues/14735#issuecomment-1594923925 ("Before migration").

**Comment by trasher** [comment]
OK It seems there is an issue on "last inventory update" migration. "last contact" and "last boot" are both OK.
I do not know what the problem is exactly, but this is not a GLPI core issue; since data migration is done from glpiinventory plugin; I'll open an issue plugin side.
